### PR TITLE
[14.0][ADD] stock_picking_cancel_reason

### PR DESCRIPTION
**Pull request by PrapassornS**
- This module is similar to purchase_cancel_reason and sale_cancel_reason , But  use for cancel stock picking

### CODE REVIEW

**Comment by dsolanki-initos** [comment]
@PrapassornS Please follow OCA guideline
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0

**Review by newtratip — APPROVED** [review]
Functional/Technical Review :+1:

**Comment by newtratip** [comment]
@dsolanki-initos, All fixed. Could you review again ?

File: stock_picking_cancel_reason/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 ProThai Technology Co.,Ltd. (http://prothaitechnology.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Stock Picking Cancel Reason",
+    "version": "14.0.1.0.0",
+    "author": "Prothai, Odoo Community Association (OCA)",
+    "category": "Stock",
+    "license": "AGPL-3",
+    "complexity": "normal",
```
Comment (kittiu) [inline stock_picking_cancel_reason/__manifest__.py]: This should be easy on. I think this flag not very useful.
```suggestion
```

File: stock_picking_cancel_reason/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 ProThai Technology Co.,Ltd. (http://prothaitechnology.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Stock Picking Cancel Reason",
+    "version": "14.0.1.0.0",
+    "author": "Prothai, Odoo Community Association (OCA)",
```
Comment (kittiu) [inline stock_picking_cancel_reason/__manifest__.py]:
```suggestion
    "author": "Ecosoft, Odoo Community Association (OCA)",
```

Comment (newtratip) [reply stock_picking_cancel_reason/__manifest__.py]: Fixed.

File: stock_picking_cancel_reason/model/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 ProThai Technology Co.,Ltd. (http://prothaitechnology.com)
```
Comment (kittiu) [inline stock_picking_cancel_reason/model/stock_picking.py]: Please change all copy right.

File: stock_picking_cancel_reason/model/stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        readonly=True,
+        ondelete="restrict",
+    )
+    description = fields.Text(string="Description", readonly=True)
+
+
+class StockPickingCancelReason(models.Model):
+    _name = "stock.picking.cancel.reason"
+    _description = "Stock Picking Cancel Reason"
+
```
Comment (kittiu) [inline stock_picking_cancel_reason/model/stock_picking.py:24]: I think if we have reason, we should also have `active` flag.
As reasons can't be removed, but we may not want to use it in the future.

File: stock_picking_cancel_reason/data/stock_picking_cancel_reason.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <record id="cancel_reason_no_needed" model="stock.picking.cancel.reason">
+        <field name="name">no longer needed</field>
```
Comment (kittiu) [inline stock_picking_cancel_reason/data/stock_picking_cancel_reason.xml]:
```suggestion
        <field name="name">No longer needed</field>
```

File: stock_picking_cancel_reason/model/stock_picking.py (hunk 4 above)
Comment (kittiu) [reply stock_picking_cancel_reason/model/stock_picking.py:24]: Just a simple "active" field will do.

**Review by kittiu — APPROVED** [review]

**Comment by kittiu** [comment]
@newtratip Functional / Code Rview, LGTM.
Please fix according to review also. Thanks!

File: stock_picking_cancel_reason/__manifest__.py (hunk 1 above)
Comment (newtratip) [reply stock_picking_cancel_reason/__manifest__.py]: Fixed.

File: stock_picking_cancel_reason/model/stock_picking.py (hunk 3 above)
Comment (newtratip) [reply stock_picking_cancel_reason/model/stock_picking.py]: Fixed.

File: stock_picking_cancel_reason/model/stock_picking.py (hunk 4 above)
Comment (newtratip) [reply stock_picking_cancel_reason/model/stock_picking.py:24]: Fixed.

File: stock_picking_cancel_reason/data/stock_picking_cancel_reason.xml (hunk 5 above)
Comment (newtratip) [reply stock_picking_cancel_reason/data/stock_picking_cancel_reason.xml]: Fixed.

**Comment by newtratip** [comment]
@kittiu, Fixed for your all comment. Thank you for your review :)

File: stock_picking_cancel_reason/views/stock_picking_views.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                <field name="name" />
+            </tree>
+        </field>
+    </record>
+
+    <record id="view_stock_picking_cancel_reason_search" model="ir.ui.view">
+        <field name="name">Stock Picking Cancel Reason</field>
+        <field name="model">stock.picking.cancel.reason</field>
+        <field name="arch" type="xml">
+            <search string="Stock Picking Cancel Reasons">
```
Comment (rousseldenis) [inline stock_picking_cancel_reason/views/stock_picking_views.xml:72]: @PrapassornS Could you fix this (runbot):

image

Comment (newtratip) [reply stock_picking_cancel_reason/views/stock_picking_views.xml:72]: Fixed.

**Pull request merged**